webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/nodes/NodeInfoService.scala

```diff
@@ -303,19 +416,139 @@ trait NodeInfoServiceCached extends NodeInfoService with NamedZioLogger with Cac
      * Get all relevant info from backend along with the
      * date of the last modification
      */
-    def getDataFromBackend(lastKnowModification: DateTime): IOResult[LocalNodeInfoCache] = {
+    def getUpdatedDataFromBackend(lastKnowModification: DateTime): IOResult[LocalNodeInfoCache] = {
+
+      final case class UpdatedNodeEntries(
+         deleted: Seq[LDAPEntry]
+       , updated: Seq[LDAPEntry]
+      )
+
+      def getUpdatedDataIO(lastKnowModification: DateTime): IOResult[UpdatedNodeEntries] = {
+        for {
+          con     <- ldap
+          t0      <- currentTimeMillis
+          deleted <- con.search(removedDit.NODES.dn, One, AND(IS(OC_NODE), GTEQ(A_MOD_TIMESTAMP, GeneralizedTime(lastKnowModification).toString)), A_MOD_TIMESTAMP , "entryCSN")
+          t1      <- currentTimeMillis
+          updated  <- getNewNodeInfoEntries(con, lastKnowModification, searchAttributes)
+          t2      <- currentTimeMillis
+        } yield {
+          TimingDebugLogger.debug(s"Getting updated node info data from LDAP: ${t2-t0}ms total (${t1-t0}ms for removed nodes, ${t2-t1} for accepted inventories)")
+          UpdatedNodeEntries(deleted, updated)
+        }
+      }
+
+      def buildUpdatedCache(updatedEntries: UpdatedNodeEntries): LocalNodeInfoCache = {
```

Comment:
again, this one and `def getUpdatedCache(allNodeEntries: AllNodeEntries): IOResult[LocalNodeInfoCache]` should be merged